lib/index.js

```diff
@@ -322,17 +325,28 @@ api.update = brCallbackify(async ({actor, id, patch, sequence}) => {
   const record = await api.get({actor, id});
 
   if(record.meta.sequence !== sequence) {
-    return new BedrockError(
+    throw new BedrockError(
       'Could not update Account. Record sequence does not match.',
       'InvalidStateError', {
         httpStatusCode: 409,
         public: true,
         actual: sequence,
         expected: record.meta.sequence
-      });
+      }
+    );
   }
-
-  const errors = jsonpatch.validate(patch, record.account);
+  const customValidate = (operation, index, tree, existingPath) => {
+    jsonpatch.validator(operation, index, tree, existingPath);
+    const pathId = /\/id$/i.test(existingPath);
+    if(pathId) {
+      throw new jsonpatch.JsonPatchError(
```

Comment:
Interesting ... will have to think about this pattern for handling patch validation.

cc: @davidlehn, @mattcollier, thoughts?

Reply:
open to alternatives, but we could look into expanding the json validator with more cases.

Reply:
I do understand the hesitation I mean I am overriding the default validator etc.

Reply:
also might want it to throw a bedrock validator.

Reply:
> also might want it to throw a bedrock error.

Well, I think it will fall out and become one so that's perhaps not necessary. I do want feedback from @collier and @davidlehn here ... guys?

Reply:
Right, if this customValidator throws, it's going to get caught here and wrapped in a BedrockError right? https://github.com/digitalbazaar/bedrock-account/blob/4d4abc83a5a998a903f0c60149e75609840e5528/lib/index.js#L350-L358

Reply:
is this good to go or not? @gannan08 has not weighed in.

Reply:
I think we're going to accept it for now and reconsider use as we find a need to do more special customization. So we're good in this PR.